include/sundials/sundials_stepper.h

```diff
@@ -0,0 +1,94 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNDIALS_STEPPER_H
+#define _SUNDIALS_STEPPER_H
+
+#include <sundials/sundials_core.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef _SUNDIALS_STRUCT_ SUNStepper_s* SUNStepper;
+
+typedef int (*SUNStepperAdvanceFn)(SUNStepper stepper, sunrealtype t0,
+                                   sunrealtype tout, N_Vector y,
+                                   sunrealtype* tret, int* stop_reason);
+
+typedef int (*SUNStepperOneStepFn)(SUNStepper stepper, sunrealtype t0,
+                                   sunrealtype tout, N_Vector y,
+                                   sunrealtype* tret, int* stop_reason);
+
+typedef int (*SUNStepperTryStepFn)(SUNStepper stepper, sunrealtype t0,
+                                   sunrealtype tout, N_Vector y,
+                                   sunrealtype* tret, int* stop_reason);
```

Comment:
How would someone set `t0` to begin with? Are you assuming this would be done with the package create/init routines before it is wrapped as a stepper?

Reply:
I would expect this to be done with the stepper `Create` or `Init` function or alternatively `Reset` could be called in the first step if we don't require a stepper to be initialized with the starting state.

Reply:
> I would expect this to be done with the stepper Create or Init

Do you mean the package "steppers"? The `SUNStepper` create does not currently have `t0` (but it could) and it has no `Init` function. 

> alternatively Reset could be called in the first step if we don't require a stepper to be initialized with the starting state

So we would assume `tR = t0 = 0` then?

Reply:
> Do you mean the package "steppers"? The `SUNStepper` create does not currently have `t0` (but it could) and it has no `Init` function.

Yes, I was thinking the constructor for an implementation of a `SUNStepper` (i.e., `MyStepper(t0, y0, ctx)`) could take the initial condition. The package steppers are a little different since they wrap an integrator that was already created with the initial condition set. Alternatively having `SUNStepper_Init(t0, y0)` of `SUNStepper_Init(t0, y0, yp0)` might not be a bad idea.

> So we would assume `tR = t0 = 0` then?

Just `tR = t0` since `t0` is not necessarily `0`

Reply:
> Just tR = t0 since t0 is not necessarily 0

Well, yes, but my point was that if there was no `Init` that took `t0` (or `Create`) then we would not know `t0`. 

>The package steppers are a little different since they wrap an integrator that was already created with the initial condition set.

CVODE/IDA do not take the initial condition and `t0` until their `Init` routines are called.  So you could have the case where a stepper is created with a integrator that does not yet have the IC/t0 set. With that in mind, I think having ` SUNStepper_Init` routine is cleaner.

Reply:
> CVODE/IDA do not take the initial condition and `t0` until their `Init` routines are called. So you could have the case where a stepper is created with a integrator that does not yet have the IC/t0 set. With that in mind, I think having ` SUNStepper_Init` routine is cleaner.

I think have an `Init` function is reasonable. As far as wrapping CVODE or IDA as a stepper, creating a stepper from a CVODE / IDA instance that was not initialized would be a user-error as the RHS / RES function would also not be set.

Reply:
I think that David's alternate option of each consuming package calling `Reset` is perhaps the most appropriate.  This already occurs in MRIStep (or it would take a minimal change to do so), and I anticipate that the operator-splitting stepper would also want to control the initial condition for any sub-stepper.